pkg/webhook/statefulset/statefulset.go

```diff
@@ -125,12 +124,18 @@ func getStsAttributes(data []byte, apiVersion string) (*metav1.ObjectMeta, *int3
 		if _, _, err := deserializer.Decode(data, nil, &set); err != nil {
 			return nil, nil, err
 		}
-		return &(set.ObjectMeta), set.Spec.UpdateStrategy.RollingUpdate.Partition, nil
+		if set.Spec.UpdateStrategy.RollingUpdate != nil {
+			return &(set.ObjectMeta), set.Spec.UpdateStrategy.RollingUpdate.Partition, nil
+		}
+		return &(set.ObjectMeta), nil, nil
 	}
 
 	set := apps.StatefulSet{}
 	if _, _, err := deserializer.Decode(data, nil, &set); err != nil {
 		return nil, nil, err
 	}
-	return &(set.ObjectMeta), set.Spec.UpdateStrategy.RollingUpdate.Partition, nil
+	if set.Spec.UpdateStrategy.RollingUpdate != nil {
+		return &(set.ObjectMeta), set.Spec.UpdateStrategy.RollingUpdate.Partition, nil
+	}
+	return &(set.ObjectMeta), nil, nil
```

Comment:
the code seems identical no matter what apiVersion is?

Reply:
updated.